### PR TITLE
Update how-to-choose-identity-solution.md

### PR DESCRIPTION
**Pull request by Rick-Anderson**
@JeremyLikness  let @tdykstra  take a quick pass and then you can review/sign-off. @tdykstra  is the best editor. Awesome content, I'm just making style guide changes. No future tense. Keep sentences short. Shorter is always better.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/security/how-to-choose-identity-solution.md](https://github.com/dotnet/AspNetCore.Docs/blob/45588b638124259843bf7cd8574a280a4fb057fc/aspnetcore/security/how-to-choose-identity-solution.md) | [Choose an identity management solution](https://review.learn.microsoft.com/en-us/aspnet/core/security/how-to-choose-identity-solution?branch=pr-en-us-29964) |

### CODE REVIEW

File: aspnetcore/security/how-to-choose-identity-solution.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ uid: security/how-to-choose-identity
...
+Most web apps support authentication to ensure that users are who they claim to be. A "user" might be a person or another app. Management of access ensures users are only able to see and modify the information they are authorized to see and modify. For example, an end user should not have access to the administrative section of a website. Identity management solutions are built to handle the requirements of authentication and authorization-related tasks. To learn more about identity management, see [What is identity and access management?](/azure/active-directory/fundamentals/introduction-identity-access-management)). There are many [identity management solutions for .NET web apps](xref:security/identity-management-solutions) available, each with different capabilities and requirements to use or install. This article provides guidance on how to choose the right solution.
```
Comment (Rick-Anderson) [inline aspnetcore/security/how-to-choose-identity-solution.md:13]:
```suggestion
Most web apps support authentication to ensure that users are who they claim to be. A *user* might be a person or another app. Management of access ensures users are only able to see and modify the information they are authorized to see and modify. For example, an end user shouldn't have access to the administrative section of a website. Identity management solutions are built to handle the requirements of authentication and authorization-related tasks. To learn more about identity management, see [What is identity and access management?](/azure/active-directory/fundamentals/introduction-identity-access-management)). There are many [identity management solutions for .NET web apps](xref:security/identity-management-solutions) available, each with different capabilities and requirements to use or install. This article provides guidance on how to choose the right solution.
```

File: aspnetcore/security/how-to-choose-identity-solution.md [hunk 2]
```diff
@@ -22,38 +22,64 @@ ASP.NET Core ships with a built-in authentication provider: [ASP.NET Core Identi
...
+A common reason a token server is requires is for applications that exposes web-based APIs that are consumed by other apps. For exposed web-based APIs, client UIs such as Single Page Applications (SPA), mobile clients, and desktop clients are considered to be part of the same app. SPA examples include Angular, React, and Blazor WebAssembly. If apps other than your web app or any client UIs must make a secure API call to your app, you'll likely want to use tokens. <!-- rewrite and shorten that sentence --> If you only have client UIs, ASP.NET Core Identity provides the option to acquire a token during authentication. The authentication token:
```
Comment (Rick-Anderson) [inline aspnetcore/security/how-to-choose-identity-solution.md:45]:
```suggestion
A common reason a token server is required is for applications that exposes web-based APIs that are consumed by other apps. For exposed web-based APIs, client UIs such as Single Page Applications (SPA), mobile clients, and desktop clients are considered to be part of the same app. SPA examples include Angular, React, and Blazor WebAssembly. If apps other than your web app or any client UIs must make a secure API call to your app, you'll likely want to use tokens. <!-- rewrite and shorten that sentence --> If you only have client UIs, ASP.NET Core Identity provides the option to acquire a token during authentication. The authentication token:
```

File: aspnetcore/security/how-to-choose-identity-solution.md [hunk 3]
```diff
@@ -22,38 +22,64 @@ ASP.NET Core ships with a built-in authentication provider: [ASP.NET Core Identi
...
+A common reason a token server is required is for applications that exposes web-based APIs that are consumed by other apps. For exposed web-based APIs, client UIs such as Single Page Applications (SPA), mobile clients, and desktop clients are considered to be part of the same app. SPA examples include Angular, React, and Blazor WebAssembly. If apps other than your web app or any client UIs must make a secure API call to your app, you'll likely want to use tokens. <!-- rewrite and shorten that sentence --> If you only have client UIs, ASP.NET Core Identity provides the option to acquire a token during authentication. The authentication token:
```
Comment (Rick-Anderson) [inline aspnetcore/security/how-to-choose-identity-solution.md:45]:
```suggestion
A common reason a token server is required is for applications that exposes web-based APIs that are consumed by other apps. For exposed web-based APIs, client UIs such as Single Page Applications (SPA), mobile clients, and desktop clients are considered to be part of the same app. SPA examples include Angular, React, and Blazor WebAssembly. If apps other than your web app or any client UIs must make a secure API call to your app, you'll likely want to use tokens.  If you only have client UIs, ASP.NET Core Identity provides the option to acquire a token during authentication. The authentication token:
```

**Review by tdykstra — APPROVED** [review]
Looks good, I left a few suggestions and a question

File: aspnetcore/security/how-to-choose-identity-solution.md [hunk 4]
```diff
@@ -22,38 +22,64 @@ ASP.NET Core ships with a built-in authentication provider: [ASP.NET Core Identi
...
+* Sign in once with a web app's form.
+* Use the resulting credentials to authenticate with other apps without having to sign-in again or choose a different password.
 
-## Decide where user data such as logins will be stored
+A token server is typically preferred to provide a secure and scalable solution for single sign on.
 
-Another important factor to consider is where the user login data will be stored. Many developers choose external, cloud-based services like Microsoft Entra ID to manage identity because the service provider takes on the responsibilities of securely storing data, keeping the software up to date with the latest security patches and releases, and complying with privacy regulations. Others prefer to store data on their own servers due to regulations, compliance, policy, and other reasons.
+For apps that don't have either of these requirements, the simplest way to quickly secure an app is to use the built-in ASP.NET Core Identity provider. Otherwise, a token server provided by a third-party identity management solution is needed. Token servers are available as:
```
Comment (Rick-Anderson) [inline aspnetcore/security/how-to-choose-identity-solution.md:57]:
```suggestion
For apps that don't exposes web-based APIs that are consumed by other apps and don't share logins with other apps, the simplest way to quickly secure an app is to use the built-in ASP.NET Core Identity provider. Otherwise, a token server provided by a third-party identity management solution is needed. Token servers are available as:
```

File: aspnetcore/security/how-to-choose-identity-solution.md [hunk 5]
```diff
@@ -65,7 +91,7 @@ The following table lists some of the things to consider when choosing your iden
 
 |**Feature**  |Pro/con|**Self-host**  |Pro/con|**Container**  |Pro/con|**Cloud**  |
```
[the remark this answers is not in the document]
Comment (Rick-Anderson) [reply aspnetcore/security/how-to-choose-identity-solution.md:92]: > 



> I find this table format hard to follow. I think you could remove the Pro/Con columns and just let the text speak for itself.

I agree. How about I move the ✅ and ❌ to the `Self-host` and `Container` column.

**Comment by Rick-Anderson** [comment]
@tdykstra  take a look at [Choose an identity management solution](https://review.learn.microsoft.com/en-us/aspnet/core/security/how-to-choose-identity-solution?branch=pr-en-us-29964) where I duplicated the table but removed the pro/con. I think it's much better.

File: aspnetcore/security/how-to-choose-identity-solution.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (JeremyLikness) [reply aspnetcore/security/how-to-choose-identity-solution.md:92]: Looks great with them removed. I agree.

**Review by JeremyLikness — APPROVED** [review]
All proposed changes look good to me

File: aspnetcore/security/how-to-choose-identity-solution.md [hunk 6]
```diff
@@ -65,18 +91,25 @@ The following table lists some of the things to consider when choosing your iden
...
 |**Customization**|✅|Self-host solutions are usually highly customizable, including code-based changes|❌|Although containerized solutions provide extensibility options, they are often more limited|❌|Cloud-based services allow customization but it is typically limited to configuration-based changes|
 |**Maintenance**|❌|Installed products require a dedicated resource to ensure all security patches are applied in a timely fashion and to manage upgrades|❌|The upgrade and patch process for containers is usually lower-friction and involves simply installing the provided container image|✅|The service provider maintains their cloud-based solution, including applying needed patches and handling upgrades|
 |**User credentials storage**|❌|You are responsible for data governance and handling breaches|❌|You are responsible for data governance and handling breaches|✅|The risks and regulations associated with handling user credentials are delegated to the service provider|
 
+|**Feature**  |**Self-host**  |**Container**  |**Cloud**  |
+|-------------:|:-:|---------------|:-:|---------------|:-:|-----------|
```
Comment (tdykstra) [inline aspnetcore/security/how-to-choose-identity-solution.md:101]:
```suggestion
|**Feature**  |**Self-host**  |**Container**  |**Cloud** |
|--------------|---------------|----------------|------------|
```

**Comment by tdykstra** [comment]
> @tdykstra take a look at [Choose an identity management solution](https://review.learn.microsoft.com/en-us/aspnet/core/security/how-to-choose-identity-solution?branch=pr-en-us-29964) where I duplicated the table but removed the pro/con. I think it's much better.

Yes but all text should be left-justified - I just submitted a suggestion to do that.
